app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -332,13 +332,33 @@ private void updateBookmarkState(MenuItem item) {
 
     public void showImage(int i, boolean isWikipediaButtonDisplayed) {
         this.isWikipediaButtonDisplayed = isWikipediaButtonDisplayed;
-        Handler handler =  new Handler();
-        handler.postDelayed(() -> pager.setCurrentItem(i), 5);
+        settingViewPagerCurrentItem(i);
     }
 
     public void showImage(int i) {
-        Handler handler =  new Handler();
-        handler.postDelayed(() -> pager.setCurrentItem(i), 5);
+        settingViewPagerCurrentItem(i);
+    }
+
+  /**
+   * This function is called to check if the selected item(item to be displayed) is loaded in the
+   * adapter or not, and when its done in background so that it does not block the ui if item is
+   * removed
+   * @param i current item that to be shown
```

Comment:
I think this solution won't work because we don't use MediaDetailPagerFragment only for contribution list. We use it in explore, bookmarks and searchFragments too. So keeping an index is a too hacky and patchy way to solve this issue which would highly cause some problems for us. So I suggest an overall strategy change for this issue. To start we can discover the root cause of this issue.

Reply:
yes 
exactly 
i have done that just waiting for your reply to make commit

Reply:
I don't know like who does that , someone has called the function made it wait for 5millisec and in mean time applied the adapter, like simply just apply the adapter one line above all problem solved , same can is happening in MediaDetailPagerFragment and 
![image](https://user-images.githubusercontent.com/56196007/114071485-ccac8a80-98be-11eb-960b-900a99e790dd.png)
luckily we can see that this used in 5 places probably with same error so those screen should be also flickering some times , that would be solved to